src/main/java/com/box/sdk/URLTemplate.java

```diff
@@ -18,12 +21,45 @@ public URLTemplate(String template) {
     }
 
     /**
-     * Build a URL with URL Parameters.
+     * Build a URL with numeric URL Parameters.
      * @param base base URL
      * @param values URL parameters
      * @return URL
      */
     public URL build(String base, Object... values) {
+        for (Object value: values) {
+            String valueString = String.valueOf(value);
+            Boolean test = NUMERIC.matcher(valueString).matches();
+            if (!NUMERIC.matcher(valueString).matches()) {
+                assert false : "An invalid path parameter passed in. It must be numeric.";
+            }
+        }
+        String urlString = String.format(base + this.template, values);
+
+        URL url = null;
+        try {
+            url = new URL(urlString);
+        } catch (MalformedURLException e) {
+            assert false : "An invalid URL template indicates a bug in the SDK.";
```

Comment:
```suggestion
            assert false : "A valid URL could not be constructed from the provided parameters.";
```

Reply:
I don't think this is true, now that we have the new sanitization logic.  If there's a malformed URL, it's due to the parameters' values, right?

Reply:
Also fix the other three instances of this verbiage in this class.